src/main/scala/za/co/absa/loginsvc/athena/PropertiesLoginServiceProfileCredentialsProvider.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.athena
+
+import com.amazonaws.auth.{AWSCredentials, AWSCredentialsProvider}
+import za.co.absa.loginsvc.athena.PropertiesLoginServiceProfileCredentialsProvider.LoginServiceProperties
+import za.co.absa.loginsvc.athena.ctx.DriverContext
+
+import java.util.Properties
+
+object PropertiesLoginServiceProfileCredentialsProvider {
+  case class LoginServiceProperties(
```

Comment:
As this is also used in the other provider, consider moving this out of the companion object to a separate file, perhaps.